adapter/internal/ingress/client_manager.go

```diff
@@ -23,9 +23,10 @@ type connection struct {
 type ClientManager struct {
 	connector     ConnectionBuilder
 	connectionTTL time.Duration
-	connections   []unsafe.Pointer
+	connections   []*connection
 	nextIdx       uint64
 	retryWait     time.Duration
+	mu            sync.RWMutex
```

Comment:
It is a Go convention to have the `mu` defined in the struct above the fields it is guarding. Not something you need to fix but for future reference.